twake/frontend/src/app/services/Apps/Messages/MessageReactionService.ts

```diff
@@ -0,0 +1,90 @@
+import { Message, ReactionType } from 'app/models/Message';
+import UserService from 'app/services/user/UserService';
+import MessageEditorManager from 'app/services/Apps/Messages/MessageEditorServiceFactory';
+import DepreciatedCollections from 'app/services/Depreciated/Collections/Collections.js';
+
+/**
+ * Service that allow you to manage message reactions
+ */
+class MessageReactionService {
+  collection: typeof DepreciatedCollections;
+
+  constructor() {
+    this.collection = DepreciatedCollections.get('messages');
+  }
+
+  /**
+   * Get user reactions in a message
+   * @param message Message
+   * @param userId string
+   */
+  public getUserReactions(message: Message, userId?: string): ReactionType[] {
+    const { reactions } = message;
+    const userReactions: ReactionType[] = [];
+
+    if (reactions?.length && userId) {
+      reactions.forEach(r => (r.users.includes(userId) ? userReactions.push(r) : false));
+    }
+
+    return userReactions;
+  }
+
+  /**
+   * Update reactions in message object
+   * @param message Message
+   * @param reactionName
+   * @param userId
+   */
+  private realtimeReactionsUpdate(message: Message, reactionName: string, userId: string): void {
+    message.reactions = message.reactions?.map(r =>
+      r.name === reactionName
+        ? { ...r, count: !r.users.includes(userId) ? r.count++ : r.count-- }
```

Comment:
You can avoid the double negation: `r.users.includes(userId) ? r.count-- : r.count++` Also as this is an approximation, you could make sure we don't return negative values. And by the way ++ / -- can have not good effects here (because you are doing this: `count = (count = count + 1)` when doing `count = count++`)
Propostiion: `count: Math.max(0, r.users.includes(userId) ? r.count-1 : r.count+1)`

Reply:
Done, thank you 👍🏻